File: java/src/main/java/dojo/supermarket/model/Product.java
package dojo.supermarket.model;

import java.util.Objects;

public record Product(String name, ProductUnit unit, String quantityType) {

    public Product(String name, ProductUnit unit) {
        this(name, unit, "");
    }

    public Product withQuantityType(String s) {
        return new Product(name, unit, s);
    }

}

File: python/domain/entities/product.py
from enum import Enum


class ProductUnit(Enum):
    EACH = 1
    KILO = 2


class Product:
    def __init__(self, name, unit):
        self.name = name
        self.unit = unit


class ProductQuantity:
    def __init__(self, product, quantity):
        self.product = product
        self.quantity = quantity
